clients/unshaded/src/main/java/tachyon/client/file/FileSystemMasterClient.java

```diff
@@ -80,285 +84,273 @@ protected void afterConnect() throws IOException {
+   * @param options method options
    * @throws IOException if an I/O error occurs
    * @throws TachyonException if a Tachyon error occurs
    */
-  // TODO(calvin): Not sure if this is necessary.
-  public synchronized List<FileBlockInfo> getFileBlockInfoList(final long fileId)
+  public synchronized void delete(final TachyonURI path, final DeleteOptions options)
       throws IOException, TachyonException {
-    return retryRPC(new RpcCallableThrowsTachyonTException<List<FileBlockInfo>>() {
+    retryRPC(new RpcCallableThrowsTachyonTException<Void>() {
       @Override
-      public List<FileBlockInfo> call() throws TachyonTException, TException {
-        return mClient.getFileBlockInfoList(fileId);
+      public Void call() throws TachyonTException, TException {
+        mClient.remove(path.getPath(), options.isRecursive());
+        return null;
       }
     });
   }
 
   /**
-   * @param fileId the file id
-   * @return a new block id for the given file id
+   * Frees a file.
+   *
+   * @param path the path to free
+   * @param options method options
    * @throws IOException if an I/O error occurs
    * @throws TachyonException if a Tachyon error occurs
    */
-  public synchronized long getNewBlockIdForFile(final long fileId)
+  public synchronized void free(final TachyonURI path, final FreeOptions options)
       throws IOException, TachyonException {
-    return retryRPC(new RpcCallableThrowsTachyonTException<Long>() {
+    retryRPC(new RpcCallableThrowsTachyonTException<Void>() {
       @Override
-      public Long call() throws TachyonTException, TException {
-        return mClient.getNewBlockIdForFile(fileId);
+      public Void call() throws TachyonTException, TException {
+        mClient.free(path.getPath(), options.isRecursive());
+        return null;
       }
     });
   }
 
   /**
-   * @return the under file system address
-   * @throws ConnectionFailedException if network connection failed
+   * @param path the URI of the file
+   * @return the list of file block information for the given file id
    * @throws IOException if an I/O error occurs
+   * @throws TachyonException if a Tachyon error occurs
    */
-  public synchronized String getUfsAddress() throws IOException, ConnectionFailedException {
-    return retryRPC(new RpcCallable<String>() {
+  public synchronized List<FileBlockInfo> getFileBlockInfoList(final TachyonURI path)
+      throws IOException, TachyonException {
+    return retryRPC(new RpcCallableThrowsTachyonTException<List<FileBlockInfo>>() {
       @Override
-      public String call() throws TException {
-        return mClient.getUfsAddress();
+      public List<FileBlockInfo> call() throws TachyonTException, TException {
+        return mClient.getFileBlockInfoList(path.getPath());
       }
     });
   }
 
   /**
-   * Creates a new file.
-   *
    * @param path the file path
-   * @param options method options
-   * @return the file id
+   * @return the file info for the given file id
```

Comment:
A Tachyon Exception will be thrown with the message that the argument must be a file.